logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/backup/RestoreBackupUseCase.kt

```diff
@@ -0,0 +1,167 @@
+                return@withContext error
+            }, { (encryptedFilePath, isPasswordProtected) ->
+                if (isPasswordProtected) {
+                    decryptExtractAndImportBackup(encryptedFilePath, extractedBackupRootPath, password!!)
+                } else {
+                    getDbPathAndImport(extractedBackupRootPath)
+                }
+            })
+        }
+
+    private suspend fun decryptExtractAndImportBackup(
+        encryptedFilePath: Path?,
+        extractedBackupRootPath: Path,
+        password: String
+    ): RestoreBackupResult {
+        val backupSource = kaliumFileSystem.source(encryptedFilePath!!)
+        val extractedBackupPath = extractedBackupRootPath / BACKUP_UNENCRYPTED_FILE_NAME
+        val backupSink = kaliumFileSystem.sink(extractedBackupPath)
+        val userIdEntity = idMapper.toCryptoModel(userId)
+        val (decodingError, backupSize) = decryptBackupFile(
+            backupSource,
+            backupSink,
+            Passphrase(password),
+            userIdEntity
+        )
+
+        if (decodingError != null) {
+            return mappedDecodingError(decodingError)
+        }
+
+        return if (backupSize > 0) {
+            // On successful decryption, we still need to extract the zip file to do sanity checks and get the database file
+            extractFiles(kaliumFileSystem.source(extractedBackupPath), extractedBackupRootPath).fold({
+                kaliumLogger.e("Failed to extract encrypted backup files")
+                Failure(BackupIOFailure("Failed to extract encrypted backup files"))
+            }, {
+                kaliumFileSystem.delete(extractedBackupPath)
+                getDbPathAndImport(extractedBackupRootPath)
+            })
+        } else {
+            Failure(RestoreBackupResult.BackupRestoreFailure.InvalidPassword)
+        }
+    }
+
+    private fun mappedDecodingError(decodingError: HeaderDecodingErrors): RestoreBackupResult = when (decodingError) {
+        INVALID_USER_ID -> Failure(InvalidUserId)
+        INVALID_VERSION -> Failure(IncompatibleBackup("The provided backup version is lower than the minimum supported version"))
+        INVALID_FORMAT -> Failure(IncompatibleBackup("The provided backup format is not supported"))
+    }
+
+    private suspend fun runSanityChecks(extractedBackupPath: Path, password: String?): Either<Failure, Pair<Path?, Boolean>> =
+        if (password.isNullOrEmpty()) {
+            // Backup is not encrypted so we don't need to return the path to the encrypted file
+            checkIsValidAuthor(extractedBackupPath).fold({ Either.Left(it) }, { Either.Right(null to false) })
+        } else {
+            // If the backup is encrypted, the sanity checks are done when decoding the file
+            checkIsValidEncryption(extractedBackupPath)
+        }
+
+    private suspend fun checkIsValidEncryption(extractedBackupPath: Path): Either<Failure, Pair<Path?, Boolean>> = with(kaliumFileSystem) {
```

Comment:
this is failing due to the extractedBackupPath pointing to the file, I have no idea what the intention was of checking if it contains the .cc20 file ? perhaps this is meant to be used within a zip file ?